Update SELF_HOSTED_SETUP.md (#37)
aws/amazon-eks-pod-identity-webhook: SELF_HOSTED_SETUP.md

--service-account-issuer
```

> ## Controller Manager configuration

Reviewer: This is incorrect and needs to be removed. The Controller Manager created keys are stored in etcd and should be separate from the TokenRequest signed keys

Author: IIRC, I couldn't get the it working without this controller manager flag. Will test again, thanks for the review, get back to you soon.

Author: Yes, I confirmed that pods with new service account tokens can't access apiserver without this controller manager flag point to the new private key file. (Minikube 1.17)

It is used by controller manager to sign service account tokens, and apiserver uses corresponding public key to verify the  token.

```
--service-account-private-key-file string
Filename containing a PEM-encoded private RSA or ECDSA key used to sign service account tokens.

From the controller-manager doc: https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/
```

I'm not sure what you mean that controller manager created keys are stored in etcd, could you elaborate more?

Reviewer: Did you specify `--service-account-key-file` only once on the API server? You'll need to specify it twice: once for the legacy service account verification, and once for the new projected service account token verification.

In regard to key flags:

Old setup
```yaml
kube-apiserver \
  --service-account-key-file=old-key.pub # So the server can validate ServiceAccount tokens signed by the old private key 

kube-controller-manager \
 --service-account-private-key-file=old-key.key # the controller manager creates ServiceAccount tokens that don't expire and are stored in etcd
```

New setup
```yaml
kube-apiserver \
  --service-account-key-file=old-key.pub # So the server can still validate ServiceAccount tokens signed by the old private key 
  --service-account-key-file=new-key.pub # So the server can validate ServiceAccount tokens signed by the new private key 
  --service-account-signing-key-file=new-key.key # So the server can create new projected ServiceAccount tokens

kube-controller-manager \
 --service-account-private-key-file=old-key.key
```

> I'm not sure what you mean that controller manager created keys are stored in etcd, could you elaborate more?

The Controller manager enforces that for every ServiceAccount there are corresponding Secret objects, and those Secrets contain a JSON Web Token (JWT) that does not expire. Secrets (and all other API objects) are stored in etcd.

The _new_ projected service account tokens are created _by the APIserver_ because the API server has the new private key (from `--service-account-signing-key-file`)

Author: @micahhausler I have read through the TokenRequest API, you are right, the section about controller manager configuration has been removed.

> You'll need to point controller manager to the new signing private key by setting the following [controller manager flags](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/).

Author: Maybe the phrasing is a bit misleading, I could change it to:

```suggestion
You'll need to the specify the [following flag](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/) with the path to the new signing private key file:
```
